compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1150,35 +1162,27 @@ object Types {
         NoType
     }
 
-    /** For a ClassInfo type, its parents,
-     *  Inherited by all type proxies. Empty for all other types.
-     *  Overwritten in ClassInfo, where parents is cached.
-     */
-    def parents(implicit ctx: Context): List[TypeRef] = this match {
-      case tp: TypeProxy => tp.underlying.parents
-      case _ => List()
-    }
-
     /** The full parent types, including all type arguments */
-    def parentsWithArgs(implicit ctx: Context): List[Type] = this match {
-      case tp: TypeProxy => tp.superType.parentsWithArgs
-      case _ => List()
+    def parents(implicit ctx: Context): List[Type] = this match {
+      case tp @ AppliedType(tycon, args) if tycon.typeSymbol.isClass =>
+        tycon.parents.map(_.subst(tycon.typeSymbol.typeParams, args)) // @!!! cache?
+      case tp: TypeRef =>
+        if (tp.info.isInstanceOf[TempClassInfo]) {
```

Comment:
Complication: Handle missing parents due to incomplete denotations here. (But it's better than to do it via suspensions as was done before).